### PR TITLE
init draft for chunk offline infer and ctc ensemble

### PR DESCRIPTION
**Pull request by Slyne**
# What does this PR do ?

This is a draft.
Please don't merge!!!

TO DO:
Test the ctc implementation.
Implement the ASR clients
Test the perf
Change the naive chunk inference in nemo
    1. change the preprocessor; instead of doing inference on one sequence at one time, we should do inference of batch of sequences together.
    2. change the preprocessor and postprocessor to do chunking inference simply using the huggingface style.

### CODE REVIEW

File: examples/asr/triton-inference-server/asr_ctc/model_repo/asr_ctc/1/model.py [hunk 1]
```diff
@@ -129,16 +125,34 @@ def initialize(self, args):
...
+        ctc_decoding = CTCDecodingConfig()
+        ctc_decoding.strategy = parameters['decoding_strategy']['string_value']
+        if ctc_decoding.strategy == 'beam_search':
+            ctc_decoding.beam_size = int(parameters['beam_size']['string_value'])
+        asr_model.change_decoding_strategy(ctc_decoding)
+        self.model = asr_model.cuda()
+        self.model.encoder.freeze()
+        self.model.decoder.freeze()
+        self.model.eval()
+        self.model.to()
```
Comment (galv) [inline examples/asr/triton-inference-server/asr_ctc/model_repo/asr_ctc/1/model.py]: Just to check, did you run this code yet? I'm not sure if `model.to()` with no argument is valid.

**Comment by Slyne** [comment]
Yes, I ran it before. Will delete it.

File: examples/asr/triton-inference-server/asr_ctc/model_repo/asr_ctc/1/model.py (hunk 1 above)
Comment (Slyne) [reply examples/asr/triton-inference-server/asr_ctc/model_repo/asr_ctc/1/model.py]: yes, i ran it. deleted this line now.

**Comment by Slyne** [comment]
Basically finished asr ctc infer (one python backend or ensemble with tensorrt engine) and offline chunk infer (nemo chunk infer and huggingface asr chunk).
Haven't tested the dataset accuracy yet.
Next step is to test accuracy and test the performance on A100.
For accuracy, we can test librispeech test-clean and test-other dataset. 
I would recommend to add a long audio file test as well.

For performance test, we need to test real dataset. However, fake data sample with fixed length could give us consistent performance. We can take it as a reference.
When sorting the test dataset and it could possibly boost the performance.
We need to come out a good benchmark for this.
We can use the same way  of calculating RTF as  the hf open asr board as well.

@galv

**Comment by Slyne** [comment]
Some other things such as adding VAD seems important for asr_ctc. I used rnn-noise and webrtc before. Would like to learn the latest VAD solution. 

Chunk inference should be able to handle long audio file.  We need some really long audio file (several hours?) to test.
